### DISCUSSION TITLE
mrs as datatype

### DISCUSSION THREAD

**Discussion by jnbusler13** [UNFmontreal/Dcm2Bids#295] [Ideas]
Hi! I wanted to suggest adding mrs as and allowable datatype. I know MRS for BIDS is not approved yet and is still listed as an extension proposal for approval but there are some software packages accepting mrs in BIDS format. Is there a way to go ahead and add this in??? Selfishly asking because I just learned how to use dcm2bids and want to also be able to do it with related mrs data :)

**Comment by arnaudbore** [comment]
I really like this idea but I never played with mrs data.
Can you share some data so I can give it a try ?

**Comment by SamGuay** [comment]
MRS is a bit trickier to deal with, it depends what type of data you have. For the conversion of data to NIfTI-MRS, I'd use [spec2nii](https://github.com/wtclarke/spec2nii) that was developed by the same people who have developed the [NifTI-MRS data standard ](https://github.com/wtclarke/mrs_nifti_standard) and FSL-MRS tool. 

Once the datasets are converted, that would be doable to reorganize into bids using dcm2bids with the mrs datatype, which should be [merged soon into the spec](https://github.com/bids-standard/bids-specification/pull/1377)!

**Comment by jnbusler13** [comment]
This is exactly what I did :) Converted with spec2nii. So I'm guessing when
MRS is merged in to BIDS that dcm2bids will then have it as an allowable
datatype?? :)

I looked at the thread. Is there a timeline for mrs to be in the BIDS spec?

I'll have to ask if I'm allowed to share the data right now but I think
there are some sample data that we could track down from one of the mrs
software tutorials and convert it to give it a go.

Best,
Jessica

On Tue, Apr 30, 2024 at 10:44 AM Samuel Guay ***@***.***>
wrote:

> MRS is a bit trickier to deal with, it depends what type of data you have.
> For the conversion of data to NIfTI-MRS, I'd use spec2nii
> <https://github.com/wtclarke/spec2nii> that was developed by the same
> people who have developed the NifTI-MRS data standard
> <https://github.com/wtclarke/mrs_nifti_standard> and FSL-MRS tool.
>
> Once the datasets are converted, that would be doable to reorganize into
> bids using dcm2bids with the mrs datatype, which should be merged soon
> into the spec
> <https://github.com/bids-standard/bids-specification/pull/1377>!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UNFmontreal/Dcm2Bids/discussions/295#discussioncomment-9276417>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXAGFI4U7RXXUCSDE33PEFLY76U57AVCNFSM6AAAAABG3KXHFGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TENZWGQYTO>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jessica N. Busler, M.P.S., Ph.D.

**Comment by arnaudbore** [comment]
dcm2bids doesn't need to wait for MRS to be merge into BIDS specification.
You can already organize your MRS data into BIDS with dcm2bids without any issue as long as you have nifti and json files.

Arnaud

**Comment by jnbusler13** [comment]
Ah I must have not done it correctly then in the configuration file. It didn’t organize my mrs files. I thought maybe it was because mrs wasn’t a data type. JessicaOn Apr 30, 2024, at 11:28, Arnaud Bore ***@***.***> wrote:﻿
dcm2bids doesn't need to wait for MRS to be merge into BIDS specification.
You can already organize your MRS data into BIDS with dcm2bids without any issue as long as you have nifti and json files.
Arnaud

**Comment by juliam98** [comment]
Hi, 

The current version of _dcm2bids_ did not generate sidecar files for mrs dicom files at all for me, so I ended up using spec2nii for that. However, the [issue](https://github.com/wtclarke/spec2nii/issues/138) I had with _spec2nii_ was that the sidecar files that _spec2nii_ generated were identical for all 3 types of mrs files (`ecc`-eddy current corrected, `noecc`-not corrected, `ref`-water reference file), so nothing I could specify in the config file criteria to distinguish them. I think this might be specific to Siemens DICOM format, where a lot of the key information is stored in a CSA headers private tags (see [here](https://nipy.org/nibabel/dicom/siemens_csa.html)), which is not accessed by the _spec2nii_ script. I was able to access them using the _dicom_parser_ package, but it's three packages for something that ideally would be done with just one. Would be great if future dcm2bids versions that support mrs files would also be able to access those private tags when creating sidecar files! 

In regards to specifications about mrs names: there is a draft version of [BIDS v1.9.10](https://bids-specification.readthedocs.io/en/bep022/).

If you will be implementing this I can share my anonymised mrs data with you if that helps!

**Reply by arnaudbore** [reply]
I can give it a try, please share your data. :+1:

**Reply by juliam98** [reply]
Thank you! I have sent you the files via email. Let me know if I can help in any way.

**Reply by arnaudbore** [reply]
Ok I took a look at your data. You could in theory use spec2nii to get your nifti and json and then use dcm2bids to sort everything in BIDS. Check this [option](https://unfmontreal.github.io/Dcm2Bids/3.1.1/how-to/use-advanced-commands/#-skip_dcm2niix).
I know the json is not really great but you could use the tag OriginalFile and use it to split your different types.
Hope it helps

**Reply by juliam98** [reply]
Hi Arnaud, 

Using the name of the file is what I currently do as a workaround with a python script and regex. 

The files could potentially be distinguished based on the `(0008, 103e) Series Description` dicom tag but for some reason the sidecar files that I get from spec2nii don't have it. I was able to access it using python (pydicom, dicom_parser), so that could work, though the tags in the CSA header would probably be most reliable - such as `WaterReferencedPhaseCorrection`, as I imagine `Series Description` is dependent on a custom label.

**Reply by arnaudbore** [reply]
There is nothing more I can do for this. I think the option would be to update spec2nii but in the meantime you can do a combination of spec2nii to convert and dcm2bids to sort in BIDS with the only tag available "OriginalFile".
